tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -4204,6 +4207,20 @@ void DeclResultIdMapper::decorateStageVarWithIntrinsicAttrs(
   decorateWithIntrinsicAttrs(decl, varInst, checkBuiltInLocationDecoration);
 }
 
+void DeclResultIdMapper::setInterlockExecutionMode(spv::ExecutionMode mode) {
+  interlockExecutionMode = mode;
+}
+
+spv::ExecutionMode DeclResultIdMapper::getInterlockExecutionMode() {
```

Comment:
Should that be an optional?
As-is, now knowing ROVs, we could understand that every declaration has by default the PixelInterlockOrdered execution mode.

Reply:
Yeah, using an optional is probably better than my current approach.